modules/kubernetes-addons/cert-manager/data.tf

```diff
@@ -0,0 +1,32 @@
+data "aws_route53_zone" "selected" {
+  for_each = contains(var.domain_names, "*") ? {} : { for domain in toset(var.domain_names) : domain => domain }
+
+  name = each.key
+}
+
+
+data "aws_iam_policy_document" "cert_manager_iam_policy_document" {
+  statement {
+    effect    = "Allow"
+    resources = ["*"]
+    actions   = ["route53:GetChange"]
+  }
+
+  dynamic "statement" {
+    for_each = contains(var.domain_names, "*") ? { "*" : "*" } : { for domain in toset(var.domain_names) : domain => data.aws_route53_zone.selected[domain].arn }
```

Comment:
```suggestion
    for_each = { for k,v in toset(var.domain_names) : k => try(data.aws_route53_zone.selected[k].arn, "*") }
```

Reply:
Wildcard is valid as far as there is correct IAM policy and it might be added with `irsa_policies` parameter. I can agree that explicit is better than implicit and in this case it is better to use explicit list of zones.